src/products/components/ProductVariantPage/form.tsx

```diff
@@ -114,13 +131,32 @@ function useProductVariantUpdateForm(
 
   const attributeInput = getAttributeInputFromVariant(variant);
   const stockInput = getStockInputFromVariant(variant);
-  const channelsInput = getChannelsInput(opts.currentChannels);
+
+  const currentChannelsWithPreorderInfo = opts.currentChannels?.map(channel => {
+    const variantChannel = variant?.channelListings?.find(
+      channelListing => channelListing.channel.id === channel.id
```

Comment:
`getById`

Reply:
`channelListing` does not implement `Node` since it nests `id` under `channel`, so `getById` is not suitable here.